pontoon/base/static/js/main.js

```diff
@@ -143,6 +143,32 @@ $(function () {
         ga('send', 'event', 'ajax', 'request', settings.url);
     });
 
+    /*
+     * Display Pontoon Add-On promotion, if:
+     *
+     * - Pontoon Add-On not installed
+     * - Page loaded on Firefox or Chrome (add-on not available for other browsers)
+     */
+    var isFirefox = navigator.userAgent.indexOf("Firefox") !== -1;
+    var isChrome = navigator.userAgent.indexOf("Chrome") !== -1;
+    var downloadLink = $('#addon-promotion').find('.get');
+    setTimeout(function() {
+        if (!window.PontoonAddon || !window.PontoonAddon.installed) {
+            if (!isFirefox && !isChrome) {
+                return;
+            }
+            if (isFirefox) {
+                downloadLink.attr('href', 'https://addons.mozilla.org/sl/firefox/addon/pontoon-tools/')
+            }
+            if (isChrome) {
+                downloadLink.attr('href', 'https://chrome.google.com/webstore/detail/pontoon-add-on/gnbfbnpjncpghhjmmhklfhcglbopagbb')
+            }
+            $('body').addClass('addon-promotion-active');
+        }
+    // window.PontoonAddon is made available by the Pontoon Add-On,
+    // but not immediatelly after the DOM is ready
+    }, 1000);
```

Comment:
@MikkCZ Is this expected?

Reply:
Not sure, probably. The extension injects it's script when the tab loading is completed, and it's all async, so who actually know when that happens.

I will try to change the way the script is injected to a little more abstract API, that would allow to inject the script earlier.

Reply:
Can you please try with the artifact from _.github/workflows/build.yml_ action of https://github.com/MikkCZ/pontoon-addon/pull/142? It should be faster.

Reply:
I tried that and couldn't access `window.PontoonAddon` from console.

Reply:
I found a bug, the Pontoon URL in the options must not end with trailing slash. I submitted a hotfix for that behaviour. I would like to use some library with validation for that eventually.

Reply:
I wasn't using a trailing slash, I used http://localhost:8000, so the new version still doesn't work.

Reply:
Don't be sorry, please. :) Which API? contentScripts?

Reply:
So, I hope I found it this time. See the pontoon-addon PR. This version I tested with some arbitrary page running on localhost, and it should work. I am not sure about the speed of the `window` object modification, but it can probably be any faster now.

Reply:
Thanks, that works now. I still need to wait for the modification to happen, though.

That said, it seems like I can drop the `setTimeout` anyways now that I have `postMessage` listener in place. At least locally, the promotional banner does not show up, even if the criteria to display it is met - because the message is sent so quickly it closes the banner.

I'll deploy to stage and see how it works there.

Reply:
I did one last change to inject the script into `<head>`, so it will hopefully get executed before any frontend JS.

Reply:
I don't see any difference, but there's definitely an improvement over the version that's currently released. The banner appears briefly with that one.

So if you ship this version today and we start the promotion on Monday... I wonder if we should keep the setTimeout in or do we expect the new version to hit users before Monday.

Reply:
It's Friday and there is a fair chance that people won't turn on their computer until Monday again. Anyway I would keep the timeout there, maybe not a full second, to avoid any blicking. That's never good and it would be happening with every page load.

Reply:
Agreed! Thanks for your help!